src/mbgl/style/sources/geojson_source.cpp

```diff
@@ -9,7 +9,7 @@
 namespace mbgl {
 namespace style {
 
-GeoJSONSource::GeoJSONSource(const std::string& id, const GeoJSONOptions& options)
+GeoJSONSource::GeoJSONSource(const std::string& id, optional<GeoJSONOptions> options)
```

Comment:
why is it optional now?

Reply:
The point of changing it is its previous signature is like` GeoJSONSource(const std::string& id, const GeoJSONOptions& = {});` So it would lead unnecessary construction of a temporary default option as it will be copied to the Impl option.